30-indoor-air-quality-and-gas-leak/README.md

```diff
@@ -0,0 +1,212 @@
+
+7. Start the program in debug mode (again, how you do this depends on the IDE: VS Code or STM32CubeIDE). In your terminal emulator, you should see something like this:
+
+```
+===================
+===================
+===== SPARROW =====
+===================
+Feb 16 2023 22:42:58
+2037335832365003001d001f
+APPLICATION HOST MODE
+CONSOLE TRACE ENABLED
+```
+
+If the ENS160 sensor is functioning correctly, you will see a message like this after a few minutes:
+
+```
+aqi: note sent. aqi 1, eco2 400, tvoc: 23
+```
+
+## Air Quality Monitoring and Alerting Behavior
+
+The firmware periodically monitors air quality and gas concentrations, and posts events with this information. Regular air quality monitoring events are not synced immediately to Notehub, and will be delivered as often as the Notecard is configured to sync with Notehub via the `hub.set` request.
+
+Air quality events are posted to the Notefile `*#aqi.qo`, where `*` is replaced with the unique ID of the Sparrow node. An air quality event has these properties:
+
+```cpp
+{
+  "app" : "nf30",       // the application name
+  "aqi" : 1,            // Air Quality Index (1-5)
+  "eco2" : 439,         // CO2 concentration in parts-per-million (ppm)
+  "tvoc" : 41           // Total volatile organic compounds (VOCs) parts-per-billion (ppb)
+}
+```
+
+Each air quality reading is checked against the [configured thresholds](#configuration). If any reading is higher than the corresponding threshold, the event is sent immediately to Notehub as an alert. Alerts are distinguished from regular monitoring events by the `alert` property.
+
+```cpp
+{
+  "alert" : 1,          // Signifies an alert
+  "app" : "nf30",       // the application name
+  "aqi" : 3,            // Air Quality Index (1-5)
+  "eco2" : 639,         // CO2 concentration in parts-per-million (ppm)
+  "tvoc" : 241          // Total volatile organic compounds (VOCs) parts-per-billion (ppb)
+}
+```
+
+The `alert` property has these values:
+
+* `alert:1`: Indicates an alert condition regarding air quality has been detected. The event is immediately synced to Notehub. You can use this to signal an alert externally, such as notifying a pager other messaging service.
+
+* `alert:2`: Signifies that the alert is still ongoing. Subsequent alerts after the initial alert have the `alert` property set to `2`, indicating that the alert is ongoing because air quality hasn't returned to normal levels. These events are not immediately sent to notehub, and are provided for continuous monitoring of air quality.
+
+* `alert:3`: Signifies a stand-down alert and that air quality has returned to normal levels. These events are sent immediately to notehub. This is typically used to send an external notification that air quality has returned to normal levels.
```

Comment:
I like this alert protocol.

Reply:
Yeah, we don't have any state management on the jsonata side of things so the firmware has to do what it can to make it easy.